Dockerfile

```diff
@@ -5,7 +5,8 @@ ENV SPARK_VERSION=2.0.2
 # install gcc
 RUN apt-get update --fix-missing && \
     apt-get install -y \
-    g++ libpython-dev libsnappy-dev
+    g++ libpython-dev libsnappy-dev \
+    build-essential libssl-dev libffi-dev
```

Comment:
Why the extra packages?

Reply:
These are dependencies for `cryptography`, see the second commit.